### ISSUE TITLE
Update/Replace bamstats' GH repo, Docker image, CI, and registries

### ISSUE THREAD

**Issue by aofarrel** [dockstore/dockstore#4945]
**Context**
It was pointed out that Getting Started with Docker does not have much information on choosing an existing Docker. In other words, to get people using Dockstore as a contributor, we are giving them an extra step (creating a Dockerfile and building a Docker image) we arguably don't need to. In a previous PR I did remove some of the complexity, such as setting up tool-only autobuilding on quay.io, but much complexity still remains. So, I was tasked with modifying Getting Started with Docker to have more information on choosing an existing Docker image was an alternative to rolling your own. In doing so, I realized that the current state of affairs around bamstats is not user-friendly, and will likely require some action beyond documentation.

**Is your feature request related to a problem? Please describe.**

Several related problems:
- The github repo we point to (except in Getting Started with WDL, where [I made my own repo](https://github.com/dockstore/wdl-bamstats) and linked to that instead due to related issues) is [our fork of collaboratory](https://github.com/dockstore/dockstore-tool-bamstats)'s version, which is confusing to navigate:
  - Why are there multiple CWLs?
  - Why is the nextflow version named differently?
  - Why is there no .dockstore.yml? 
  - Why is this a totally different set of workflows used in demonstrating how to actually register workflows?
  - I know the answers to these questions (showing different types of CWLs, Nextflow standards, it was made before app registration, because we are showing GitHub app registration in that page), but users would not.
- There are several Docker images with "bamstats" in the name that also seem official (ie have an organization name, not "joe-schmo", in the prefix) but it's not quite clear if the containers from seemingly different organizations are in any way linked
   - Getting Started with CWL/WDL/NFL all have workflows that link to [a quay.io Docker image under the collaboratory organization](quay.io/collaboratory/dockstore-tool-bamstats:1.25-6_1.0)
   - ...which is a problem, because most Dockstore users likely don't know what collaboratory is and wouldn't make the connection that something owned by collaboratory might also owned by us
   - There is [a DockerHub version of collaboratory's bamstats](https://hub.docker.com/r/collaboratory/dockstore-tool-bamstats/tags), but with totally different tags compared to quay.io's tags 
   - There's also [a biocontainer version on quay](https://quay.io/repository/biocontainers/biopet-bamstats), but it's even more mysterious. Who made it? Why does it not come up when you search "bamstats" on biocontainer's own registry? Who maintains it? Is it associated with Dockstore? Since I have been asked to rewrite our docs in such a way as to encourage people to use biocontainers, will users find this maybe-official-biocontainer-maybe-not and think it's the better version? 
- The Dockerfile we have people write implies that Brian O'Conner is still the current point-of-contact for keeping it maintained
- The Dockerfile is also based on Ubuntu 14 [which recently had its EOL extended](https://ubuntu.com/blog/ubuntu-14-04-and-16-04-lifecycle-extended-to-ten-years), but only to April 2024 (that may sound like a while away but 14 came out in 2014 and was supposed to die last April)
- Some users have reported that some versions of bamstats do not work as we present them. IIRC, the WDL version is nonfunctional, and that was why I replaced the repo entirely with my own version. But I can't recall the specifics. Testing of all versions is needed.

**Describe the solution you'd like**
- Getting Started with Docker should talk about how to choose an existing Docker image (ie an existing up to date bamstats image) as well as instructions on how to build bamstats from a Dockerfile
- Clarification of which bamstats Docker images are """official"""
- Getting Started with x workflow language should use the same existing Docker image (and tell people how to replace it with their own Docker image) as the existing Docker image we suggest in Getting Started with Docker

To that end:
1. Make a free-tier Dockstore-branded [quay.io](http://quay.io/) account (instead of reusing collaboratory)
2. Make a free-tier Dockstore-branded Docker Hub account (instead of reusing collaboratory) and/or organization
3. ~Have me remake the Docker image based on a newer version of Ubuntu and mark me (or someone else) as the maintainer~ Done! https://github.com/dockstore/dockstore-tool-bamstats/pull/3
4. Let me push the remade Docker image to our Dockstore-branded quay and Docker Hub accounts, keeping the tag(s) consistent across both platforms
5. Edit the description of https://hub.docker.com/r/collaboratory/dockstore-tool-bamstats/tags and https://quay.io/repository/collaboratory/dockstore-tool-bamstats?tab=info to note they are depreciated and to use the Dockstore-branded versions instead
6. To address the GitHub repo's confusing layout, have me do one of the following:
  - heavily modify https://github.com/dockstore/dockstore-tool-bamstats in the form of PRs (and retire https://github.com/dockstore/wdl-bamstats)
  - start fresh with at dockstore github repo focused on all three languages (and retire https://github.com/dockstore/wdl-bamstats)
  - start fresh with one dockstore github repo per language (like I did with https://github.com/dockstore/wdl-bamstats)
7. Once all that is done I'll update the links and text of the docs accordingly

**Describe alternatives you've considered**
Replace bamstats with a brand new example that is also a real tool, and developed/maintained in house, and make the requisite Docker images for it. I think this would end up being more work in the end.

**Additional context**

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/DOCK-2179)
┆fixVersions: Dockstore 1.13
┆friendlyId: DOCK-2179
┆sprint: 93- Ursula
┆taskType: Story

**Comment by denis-yuen** [comment]
> That github repo is also a little confusing to navigate; why are there multiple CWLs? Why is the nextflow version named differently? Why is there no .dockstore.yml?

Probably pre-dates github app
Multiple CWLs are demoing that you can use multiple CWLs for the same dockerfile (which is something cwl folks recommended actually) https://github.com/dockstore/dockstore-tool-bamstats/blob/develop/Dockstore2.cwl#L8

I think most nextflow repos have a `[main.nf](http://main.nf)` so I think we're just following convention, maybe I'll tackle in github/jira

**Comment by aofarrel** [comment]
Going to hold off on revamping "getting started with CWL" (TLP-2600) until we come to a consensus on this, although I think I can move forward on adding information about selecting an existing Docker image in "getting started with Docker" (TLP-2599). This issue has implications for GS w/ Docker too, but it's less central to the changes TLP-2599 is about.

**Comment by denis-yuen** [comment]
re: collaboratory namespace and Brian O'Connor as maintainer

I'm sure that we can create a new one, I don't think there's much behind "collaboratory" except that the tutorials are from then

> To that end

Seems fine except for "Let me push the remade Docker image to our Dockstore-branded quay and Docker Hub accounts, keeping the tag(s) consistent across both platforms"

I don't know that there is much benefit to having both now that we have github apps but for tools. We could just pick one

**Comment by denis-yuen** [comment]
You might also appreciate https://github.com/morgantaschuk/one-workflow-many-ways as an example of a repo with a "real" workflow in many different languages

**Comment by denis-yuen** [comment]
> I'd be open to making that our new go-to example but I'd probably want to fork it or start fresh on the Dockstore organization (pending the author's permission).

Forking, I'd have assumed so. It's MIT license so you're good.

**Comment by aofarrel** [comment]
Turns out bamstats was built on Travis, which from what I've read has become a pain for its open-source tier as you have to keep requesting credits (in spite of what the homepage implies). It may be worth migrating the repo to CircleCI at some point, but given how tricky it can be to get Cromwell (I guess we could use miniwdl instead?) running on a CI system I think that probably deserves its own ticket.

I don't consider CI to be a blocker on updating GS with CWL, so we can table that part of the discussion, but everything else here probably needs to be considered before updating GS with CWL.

**Comment by aofarrel** [comment]
Docker update: https://github.com/dockstore/dockstore-tool-bamstats/pull/3

**Comment by denis-yuen** [comment]
Suggestion for simplifying the breakdown for a Dockstore developer if assigned:
- 1) quay.io org for Dockstore exists, can be used instead of collaboratory
- 2) docker hub for Dockstore exists but may be inaccessible, may need to create something new if not collaboratory

For 3), 4), 5) work with Ash and/or Lighthouse
